graphql_compiler/cost_estimation/statistics.py

```diff
@@ -102,15 +117,24 @@ class name) to count of edge instances of given class
             distinct_field_values_counts: optional dict, (str, str) -> int, mapping vertex class
                                           name and property field name to the count of distinct
                                           values of that vertex class's property field.
+            field_quantiles: optional dict, (str, str) -> list, mapping vertex class name
+                             and property field name to a list of N quantiles, a sorted list of
+                             values seperating the values of the field into N-1 groups of equal
+                             size. The first element of the list is the smallest known value,
+                             and the last element is the largest known value. The number N can be
+                             different for each entry.
```

Comment:
Do the number of values have to be divisible by N-1 in order for the groups to be of equal size? What interpolation method should I use to compute quantiles?

(See the parameters section in the link below so that we have a common language to talk about interpolations).
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.quantile.html

Reply:
1. Exact equality is not important, the statistics are stale anyway.
2. Don't interpolate.

Reply:
Can you define the ith-quantile here then as the smallest field value that is larger of the fraction i/N of other field values? Just to make it clear that no interpolation is used, (especially since libraries often compute some sort of interpolation to evaluate quantiles).

Also, instead of writing:
```into N-1 groups of equal size```
Write:
```into N-1 groups of (possibly) equal size```